src/network/ip_address.cpp

```diff
@@ -114,6 +118,12 @@ bool mp::IPAddress::operator>=(const IPAddress& other) const
     return as_uint32() >= other.as_uint32();
 }
 
+mp::IPAddress mp::IPAddress::operator=(const IPAddress& other)
+{
+    octets = other.octets;
+    return other;
+}
+
```

Comment:
I omitted the copy assignment operator so that an instance of IPAddress is invariant.